morpheus/utils/json_schema_builder.py

```diff
@@ -0,0 +1,519 @@
+# Copyright (c) 2023-2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from morpheus.utils.column_info import ColumnInfo
+from morpheus.utils.column_info import DataFrameInputSchema
+from morpheus.utils.column_info import DateTimeColumn
+from morpheus.utils.column_info import DistinctIncrementColumn
+from morpheus.utils.column_info import IncrementColumn
+from morpheus.utils.column_info import RenameColumn
+from morpheus.utils.column_info import StringCatColumn
+from morpheus.utils.column_info import StringJoinColumn
+from morpheus.utils.column_info import BoolColumn
+from datetime import datetime
+import warnings
+
+
+class JSONSchemaBuilder:
```

Comment:
Two comments:
1. Look at converting the ColumnInfo classes to use pydantic. Then most of this logic can be contained within each of the `ColumnInfo` classes themselves. Reducing the chance they get out of sync.
2. Add a static function to DataFrameInputSchema to automatically create an instance of this class and load it from disk. `DataFrameInputSchema.from_json("some path") -> DataFrameInputSchema`